packages/backend/src/database/entities/changesets.ts

```diff
@@ -0,0 +1,87 @@
+import { Knex } from 'knex';
+import { z } from 'zod';
+
+export const ChangesetsTableName = 'changesets';
+export const ChangesTableName = 'changes';
+
+export const ChangesetStatusSchema = z.enum([
+    'draft',
+    'pending',
+    'approved',
+    'rejected',
+    'applied',
+]);
+export type ChangesetStatus = z.infer<typeof ChangesetStatusSchema>;
+
+export const ChangeTypeSchema = z.enum(['create', 'update', 'delete']);
+export type ChangeType = z.infer<typeof ChangeTypeSchema>;
+
+export const EntityTypeSchema = z.enum(['table', 'dimension', 'metric']);
+export type EntityType = z.infer<typeof EntityTypeSchema>;
+
+export const DbChangesetSchema = z.object({
+    changeset_uuid: z.string().uuid(),
+    created_at: z.date(),
+    updated_at: z.date(),
+    created_by_user_uuid: z.string().uuid(),
+    updated_by_user_uuid: z.string().uuid(),
+    project_uuid: z.string().uuid(),
+    status: ChangesetStatusSchema,
+    name: z.string().min(1),
+});
+
+export type DbChangeset = z.infer<typeof DbChangesetSchema>;
+
+export const DbChangesetInsertSchema = DbChangesetSchema.pick({
+    created_by_user_uuid: true,
+    updated_by_user_uuid: true,
+    project_uuid: true,
+    status: true,
+    name: true,
+});
+
+type DbChangesetInsert = z.infer<typeof DbChangesetInsertSchema>;
+
+export const DbChangesetUpdateSchema = DbChangesetSchema.pick({
+    updated_by_user_uuid: true,
+    status: true,
+    name: true,
+});
+
+type DbChangesetUpdate = z.infer<typeof DbChangesetUpdateSchema>;
+
+export type ChangesetsTable = Knex.CompositeTableType<
+    DbChangeset,
+    DbChangesetInsert,
+    DbChangesetUpdate
+>;
+
+export const DbChangeSchema = z.object({
+    change_uuid: z.string().uuid(),
+    changeset_uuid: z.string().uuid(),
+    created_at: z.date(),
+    created_by_user_uuid: z.string().uuid(),
+    source_prompt_uuid: z.string().uuid().nullable(),
+    type: ChangeTypeSchema,
+    entity_type: EntityTypeSchema,
+    entity_explore_uuid: z.string().uuid().nullable(),
+    entity_name: z.string().min(1),
+    payload: z.record(z.unknown()),
```

Comment:
makes sense to keep it **unknown** until we start the tool implementation or the explore<->changeset implementation no?